sql/src/main/java/io/crate/execution/engine/collect/sources/CollectSourceResolver.java

```diff
@@ -106,7 +106,7 @@ public CollectSourceResolver(ClusterService clusterService,
         this.shardCollectSource = shardCollectSource;
         this.fileCollectSource = new ProjectorSetupCollectSource(fileCollectSource, projectorFactory);
         this.tableFunctionSource = new ProjectorSetupCollectSource(tableFunctionCollectSource, projectorFactory);
-        this.emptyCollectSource = new ProjectorSetupCollectSource(new VoidCollectSource(), projectorFactory);
+        this.emptyCollectSource = new EmptyCollectSource(new VoidCollectSource());
```

Comment:
Btw `VoidCollectSource´ is already a `CollectSource`, there is no need to wrap it.

I'm also not sure if it's right to skip *all* projections. Shouldn't we instead consider to adapt the `ProjectorSetupCollectorSource` to only initialize the node projections and skip the shard projections?

This would make it consistent with the behaviour of the `ShardCollectSource` in case shards are unavailable after it has already been selected